File: frontend/src/types/health.ts
export interface HealthEvent {
  id: string
  animal_id: string
  establishment_id: string
  event_type: 'vaccine' | 'treatment' | 'disease' | 'surgery' | 'checkup' | 'deworming' | 'vitamin'
  product_name: string
  dose: string | null
  route: string | null
  event_date: string
  next_date: string | null
  vet_name: string | null
  cost: number | null
  notes: string | null
  created_at: string
}

File: backend/app/schemas/health.py
from pydantic import BaseModel
from datetime import date, datetime
import uuid
from app.models.health import HealthEventType


class HealthEventCreate(BaseModel):
    animal_id: uuid.UUID
    establishment_id: uuid.UUID
    event_type: HealthEventType
    product_name: str
    dose: str | None = None
    route: str | None = None
    event_date: date
    next_date: date | None = None
    vet_name: str | None = None
    cost: float | None = None
    notes: str | None = None


class HealthEventResponse(HealthEventCreate):
    id: uuid.UUID
    tenant_id: uuid.UUID
    created_at: datetime

    model_config = {"from_attributes": True}
